### PR TITLE
Simplify code to get GCS/GCB clients

### PR DESCRIPTION
**Pull request by dgageot**
+ Remove duplication
 + Remove indirections

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/gcp/client.go [hunk 1]
```diff
@@ -17,60 +17,22 @@ limitations under the License.
...
-var (
-	cbclient *cloudbuild.Service
-	cbOnce   sync.Once
-	cbErr    error
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
 )
 
-// CloudBuildClient returns an authenticated client for interacting with
-// the Google Cloud Build API. This client is created once and cached
-// for repeated use in Skaffold.
```
Comment (briandealwis) [inline pkg/skaffold/gcp/client.go]: So I think we'll now be creating a new client per build?

Comment (dgageot) [reply pkg/skaffold/gcp/client.go]: yes. I didn't see the value of caching the client. can only create more problems.

File: pkg/skaffold/gcp/client.go [hunk 2]
```diff
@@ -17,60 +17,22 @@ limitations under the License.
...
-func CloudBuildClient() (*cloudbuild.Service, error) {
-	cbOnce.Do(func() {
-		var options []option.ClientOption
-		var err error
-		creds, cErr := activeUserCredentials()
```
Comment (balopat) [inline pkg/skaffold/gcp/client.go]: why are we removing this? This is an important piece of logic that we implemented to fix https://github.com/GoogleContainerTools/skaffold/issues/1966 - unless I'm missing something and the client libraries include this logic now?

Comment (dgageot) [reply pkg/skaffold/gcp/client.go]: This is not removed, just moved to `ClientOptions()`

Comment (balopat) [reply pkg/skaffold/gcp/client.go]: gotcha, I missed that

**Pull request merged**